### PR TITLE
[WIP]Use product image according to client device

### PR DESCRIPTION
**Pull request by rerissondaniel**
#### What is the purpose of this pull request?

Use resized images depending on the client device. 
#### What problem is this solving?

The image that was used had always the same size, it could make its loading slow, depending on the network. 

#### How should this be manually tested?
[workspace](https://imagesizing--storecomponents.myvtex.com/)
Check that the product image size is different in mobile and desktop.

#### Types of changes
- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Requires change to documentation, which has been updated accordingly.

### CODE REVIEW

File: react/components/Image.js [hunk 1]
```diff
@@ -1,15 +1,39 @@
...
+  resizeUrl = src => {
+    //Check of src is already in a size defined by the client
+    const resized = /[1-9]+-(:?([1-9]+)|(auto))/
+    if (resized.exec(src)) {
+      return src
+    }
+
+    const { runtime: { hints: { mobile } } } = this.props
+
+    const imageId = src.slice(src.indexOf('/ids/')).split('/')[2]
```
Comment (luciannojunior) [inline react/components/Image.js]: Duplicated code from https://github.com/vtex-apps/product-details/blob/master/react/utils/generateUrl.js#L22.

@brunojdo do you agree we should have a HOC in `store-components` that executes this behavior?

File: react/components/Image.js [hunk 2]
```diff
@@ -1,15 +1,39 @@
...
-    <div
-      className={classNames(className, 'vtex-aspect-ratio w-100')}
-      style={{ backgroundImage: `url(${src})` }}
-    />
-  )
+class Image extends Component {
+
+  resizeUrl = src => {
+    //Check of src is already in a size defined by the client
+    const resized = /[1-9]+-(:?([1-9]+)|(auto))/
```
Comment (luciannojunior) [inline react/components/Image.js]: `const isWithMeasures = new Regex(...)` seems more intuitive

**Comment by luciannojunior** [comment]
Nice functionality!

File: react/components/Image.js (hunk 1 above)
Comment (brunojdo) [reply react/components/Image.js]: Add TODO, Technical Debt

File: react/components/Image.js [hunk 3]
```diff
@@ -1,15 +1,39 @@
-import React from 'react'
+import React, { Component } from 'react'
 import PropTypes from 'prop-types'
 import classNames from 'classnames'
+import { withRuntimeContext } from 'render'
+
+const MOBILE_SIZE = 200
```
Comment (brunojdo) [inline react/components/Image.js]: In my desktop the size is 273. And the mobile is 273 too. I think we can only one type like 400. We need to test more how this scale

**Pull request closed without merging**